### ISSUE TITLE
only spaces get predicted

### ISSUE THREAD

**Issue by ghost** [hkproj/pytorch-transformer#18]
So i am definitely using a different dataset and after training it for 20 epochs the output that is predicted is only spaces. this is the dataset that i used https://huggingface.co/datasets/ai4bharat/samanantar atleast 40% of it. its a relatively big dataset. hence i assume that my model has not seen the dataset enough to start predicting or is there any problem that might occur due to encoding. what are your thoughts about this? if you have time you can go through my codes as well which is in https://github.com/aadityapattabhiraman/transformers. thank you

**Comment by munanning** [comment]
I faced similar problem, and it generated strange symbol.

Processing epoch 00: 100%|____________________________________________________________________________________________________| 3638/3638 [07:18<00:00,  8.30it/s, loss=4.159]
--------------------------------------------------------------------------------
SOURCE: The old Princess, Mary Borisovna, Kitty's godmother, who had always been very fond of her, particularly wished to see her.
TARGET: La vecchia principessa Mar_ja Borisovna, madrina di Kitty, che sempre le aveva voluto molto bene, desider_ assolutamente di vederla.
PREDICTED: _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
 _ _ _ _ _ _
--------------------------------------------------------------------------------
SOURCE: 'I would maintain the opposite...' began Koznyshev.
TARGET: _ Io dico al contrario... _ cominci_ Sergej Ivanovic.
PREDICTED: _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
 _ _ _ _ _ _

**Comment by ghost** [comment]
possible explanation could be that it was not trained enough

**Comment by munanning** [comment]
I find one possible solution:
I wrote the masked_fill_ to masked_fill, then the words are not masked correctly in Decoder.

**Comment by Tyler-Hilbert** [comment]
> possible explanation could be that it was not trained enough

Are you saying the model started working after training longer? What epoch did you train to?
